actions/issue/add-to-project/entrypoint

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+set -euo pipefail
+shopt -s inherit_errexit
+
+# Github CLI does not currently allow interacting with the API of new-style projects.
+# Github GraphQL API guide: https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-api-to-manage-projects
+
+function main() {
+  declare issue_node_id project_org project_num field_name option_name
+  declare -i project_num
+
+  while [[ "${#}" != 0 ]]; do
+    case "${1}" in
+      --token)
+        token="${2}"
+        shift 2
+        ;;
+
+      --issue-node-id)
+        issue_node_id="${2}"
+        shift 2
+        ;;
+
+      --project-org)
+        project_org="${2}"
+        shift 2
+        ;;
+
+      --project-num)
+        project_num=${2}
+        shift 2
+        ;;
+
+      --field-name)
+        field_name="${2}"
+        shift 2
+        ;;
+
+      --option-name)
+        option_name="${2}"
+        shift 2
+        ;;
+
+      "")
+        # skip if the argument is empty
+        shift 1
+        ;;
+
+      *)
+        echo "Unknown argument" "$@"
+        exit 1
+    esac
+  done
+
+  echo "${token}" | gh auth login --with-token
+
+  project_id=
```

Comment:
I have a slight preference for `local project_id` rather than `project_id=` (or `project_id=""`). I think it more clearly maps onto the intention that the variable hasn't been set, rather than it was set and it's empty.

If you agree, would you be willing to update the various variables in this script?

Reply:
Done. See refactored code